index.js

```diff
@@ -380,7 +394,13 @@ FSWatcher.prototype.add = function(paths, _origAdd, _internal) {
   paths = arrify(paths);
 
   if (cwd) paths = paths.map(function(path) {
-    return isAbsolute(path) ? path : sysPath.join(cwd, path);
+    if(isAbsolute(path)){
+      return path;
+    } else if(path[0] == '!'){
```

Comment:
In all cases, the `cwd` and `path` should not be joined if the path is absolute, just like it was before.